src/stringmap.c

```diff
@@ -43,14 +43,20 @@
  */
 stringmap_t *stringmap_load(const char *filename, int numFinalWordsToMatch)
 {
-    stringmap_t *result = calloc(1, sizeof(*result));
-    FILE *fp = fopen(filename, "r");
+    stringmap_t *result;
+    FILE *fp;
     char buf[2*PATH_MAX];
     int n;
 
+    result = calloc(1, sizeof(*result));
+    if (!result)
+        return NULL;
     result->numFinalWordsToMatch = numFinalWordsToMatch;
-    if (!fp)
+    fp = fopen(filename, "r");
```

Comment:
you can try to open the file first, so that you don't have to free.

Reply:
https://github.com/distcc/distcc/commit/bbfddaa66e85913abce5f978c6b22a11d37326bc